pkg/controller/tls/certs.go

```diff
@@ -225,55 +225,64 @@ func findExistingCertSecret(ctx context.Context, client kclient.Client, target s
 }
 
 func (u *LEUser) provisionCertIfNotExists(ctx context.Context, client kclient.Client, domain string, namespace string, secretName string) error {
-	// Find existing secret if exists
+	// Find existing secret by expected name
 	existingSecret := &corev1.Secret{}
 	findSecretErr := client.Get(ctx, router.Key(namespace, secretName), existingSecret)
+	var mustUpdate bool
 	if findSecretErr == nil {
-		// Skip: TLS secret already exists, nothing to do here, renewal will be handled elsewhere
-		return nil
+		if existingSecret.Annotations[labels.AcornDomain] == domain {
+			// Skip: TLS secret already exists and matches the domain, nothing to do here, renewal will be handled elsewhere
+			logrus.Debugf("No need to provision cert for domain %s, secret for that domain already exists: %s/%s", domain, existingSecret.Namespace, existingSecret.Name)
+			return nil
+		}
+		// There's an existing secret with a different domain but matching the expected name, so we'll update it
+		logrus.Debugf("Updating existing secret %s/%s with domain %s to %s", existingSecret.Namespace, existingSecret.Name, existingSecret.Annotations[labels.AcornDomain], domain)
+		mustUpdate = true
 	} else if !apierrors.IsNotFound(findSecretErr) {
 		// Not found is ok, we'll create the secret below.. other errors are bad
-		logrus.Errorf("Error getting secret %s/%s: %v", namespace, secretName, findSecretErr)
+		logrus.Errorf("Error getting certificate secret %s/%s: %v", namespace, secretName, findSecretErr)
```

Comment:
Nit: I am a big fan of handling error at top level. If we are logging error and return error at the same time, likely the error will be logged and processed again at the top level.

Reply:
Fair point. Changed that :)